pkgs/tools/backup/s3ql/default.nix

```diff
@@ -1,32 +1,49 @@
-{ stdenv, fetchFromGitHub, python3Packages, sqlite, which }:
+{ stdenv, fetchFromGitHub, fetchpatch, makeWrapper, python3Packages, sqlite, which, psmisc }:
 
 python3Packages.buildPythonApplication rec {
   pname = "s3ql";
-  version = "3.3.2";
+  version = "3.6.0";
 
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
     rev = "release-${version}";
-    sha256 = "1x0xj8clfs8fdczn8skc2wag5i4z47bsvlczn22iaf20hll1bb2w";
+    sha256 = "04w8b15mwvyxnq146diblgs83mwsz9m0xmnsss5d1z2vz32i1q3z";
   };
 
-  checkInputs = [ which ] ++ (with python3Packages; [ cython pytest ]);
+  patches = [
+    # Fixes tests with pytest 6, to be removed in next stable version
+    (fetchpatch {
+      url = "https://github.com/s3ql/s3ql/commit/5316c60b447dbf85a8b80de523fb1a570bf01c11.patch";
+      sha256 = "142j8ffhg6w7pf8cpxj0i485fq9vhvac32zjwl9ka4r3fqcq79zb";
+    })
+  ];
+
+  nativeBuildInputs = [ makeWrapper ] ++ (with python3Packages; [
+    cython
+  ]);
+
   propagatedBuildInputs = with python3Packages; [
-    sqlite apsw pycrypto requests defusedxml dugong llfuse
-    cython pytest pytest-catchlog google_auth google-auth-oauthlib
+    sqlite apsw pycrypto requests defusedxml dugong pyfuse3
+    google_auth google-auth-oauthlib
   ];
 
   preBuild = ''
     ${python3Packages.python.interpreter} ./setup.py build_cython build_ext --inplace
   '';
 
-  checkPhase = ''
-    # Removing integration tests
-    rm tests/t{4,5,6}_*
-    pytest tests
+  postInstall = ''
+    wrapProgram $out/bin/umount.s3ql \
+      --prefix PATH : ${stdenv.lib.makeBinPath [ psmisc ]}
   '';
```

Comment:
I would prefer the source could be patched to call the related utility, as this will allow it to work even when imported or when using `python -m`